tests/Unit/NumericalAlgorithms/Interpolation/Test_LinearLeastSquares.cpp

```diff
@@ -41,6 +42,53 @@ void test_linear_least_squares(const std::array<double, Order + 1>& coeffs,
           my_approx(gsl::at(y_values, i)));
   }
 }
+
+void test_linear_least_squares_datavector() {
+  const DataVector x_values{1.0, 2.0, 3.0, 4.0, 5.0};
+  const std::vector<std::array<double, 2>> coefficients{
+      {{6.0, 7.0}}, {{8.0, 9.0}}, {{10.0, 11.0}}, {{12.0, 13.0}}};
+  std::vector<DataVector> y_values{4, DataVector{5}};
+  for (size_t i = 0; i < y_values.size(); ++i) {
+    y_values[i] = x_values * coefficients[i][1] + coefficients[i][0];
+  }
+  intrp::LinearLeastSquares<1> lls(x_values.size());
+  const std::vector<std::array<double, 2>> computed_coefficients =
+      lls.fit_coefficients(x_values, y_values);
+  CHECK_ITERABLE_APPROX(coefficients, computed_coefficients);
+  const double x_rand = 2.31;
+  for (size_t i = 0; i < y_values.size(); ++i) {
+    double expected_y_value = x_rand * coefficients[i][1] + coefficients[i][0];
+    double computed_y_value = lls.interpolate(coefficients[i], x_rand);
+    CHECK(expected_y_value == computed_y_value);
+  }
+}
+
+template <size_t Order>
+void test_linear_least_squares_datavector2(
+    const std::vector<std::array<double, Order + 1>>& coeffs,
+    const DataVector& x_values) {
+  std::vector<DataVector> y_values{coeffs.size(),
+                                   DataVector{x_values.size(), 0.0}};
+  for (size_t i = 0; i < y_values.size(); ++i) {
+    for (size_t j = 0; j < Order + 1; j++) {
```

Comment:
Can't this inner loop be replaced by `evaluate_polynomial` in `Utilities/Math.hpp`?